client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -68,48 +79,207 @@ static Request get(GetRequest getRequest) {
     }
 
     private static Map<String, String> getParams(GetRequest getRequest) {
-        Map<String, String> params = new HashMap<>();
-        putParam("preference", getRequest.preference(), params);
-        putParam("routing", getRequest.routing(), params);
-        putParam("parent", getRequest.parent(), params);
-        if (getRequest.refresh()) {
-            params.put("refresh", Boolean.TRUE.toString());
+        Params parameters = Params.builder();
+        parameters.withPreference(getRequest.preference());
+        parameters.withRouting(getRequest.routing());
+        parameters.withParent(getRequest.parent());
+        parameters.withRefresh(getRequest.refresh());
+        parameters.withRealtime(getRequest.realtime());
+        parameters.withStoredFields(getRequest.storedFields());
+        parameters.withVersion(getRequest.version());
+        parameters.withVersionType(getRequest.versionType());
+        parameters.withFetchSourceContext(getRequest.fetchSourceContext());
+        return parameters.getParams();
+    }
+
+    private static String getEndpoint(GetRequest getRequest) {
```

Comment:
Yes we can. I'm also waiting for Luca opinion on those Endpoint/Params classes, but if he likes it I'll move those as well.